### PR TITLE
Documentation added - lora driver sample - send.

### PR DESCRIPTION
**Pull request by Spice-Weasel**
I only have a single lora device available, so have tested the send on this with no receiver. I have built and flashed the firmware onto a STMicroelectronics Discovery Lora Devkit (b_l072z_lrwan1).

I have observed the console output and documented this. When my next paycheque comes in, I will document the receiver.

### CODE REVIEW

File: samples/drivers/lora/send/README.rst
Comment (mniestroj) [inline samples/drivers/lora/send/README.rst:9]: Please squash this change into first commit.

File: samples/drivers/lora/send/README.rst
Comment (mniestroj) [inline samples/drivers/lora/send/README.rst:37]: Please just leave single newline at the end of file.

File: samples/drivers/lora/send/README.rst
Comment (mniestroj) [inline samples/drivers/lora/send/README.rst:36]: Still one newline too much, which compliance CI check is complaining about.

File: samples/drivers/lora/send/README.rst
Comment (mniestroj) [inline samples/drivers/lora/send/README.rst:33]: I guess there needs to be a newline betwee `.. code-block:: console` and first log line. It looks like this is the reason why documentation build failed.

**Review by mniestroj — COMMENTED** [review]
Please take a look at compliance checks. You need to use the same name+email in git commit author as in signed-off-by entry in the commit body.

File: samples/drivers/lora/send/README.rst
Comment (Spice-Weasel) [reply samples/drivers/lora/send/README.rst:36]: Thanks, it must be case sensitive as I have not capitalized the first letters of the names in git.

File: samples/drivers/lora/send/README.rst
Comment (JordanYates) [inline samples/drivers/lora/send/README.rst:20]: `where you board` -> `where your board`

Comment (JordanYates) [inline samples/drivers/lora/send/README.rst:20]: > default lora radio

I think it is worth briefly expanding on what this means (has a `lora0` alias in devicetree)

**Review by JordanYates — CHANGES_REQUESTED** [review]

File: samples/drivers/lora/send/README.rst
Comment (Spice-Weasel) [reply samples/drivers/lora/send/README.rst]: Understood, I have added your suggestion.

File: samples/drivers/lora/send/README.rst
Comment (kartben) [inline samples/drivers/lora/send/README.rst:5]: Several suggestions below:

- name the sample so that it's easier to add lora-receive later on
- Spell "LoRa", not "lora"
- `relevant-api` is meant to be a list of existing Doxygen groups.

```suggestion
.. zephyr:code-sample:: lora-send
   :name: LoRa send
   :relevant-api: lora_api

   Transmit a preconfigured payload every second using the lora radio.
```

File: samples/drivers/lora/send/README.rst
Comment (kartben) [inline samples/drivers/lora/send/README.rst:14]:
```suggestion
This sample demonstrates how to use the LoRa radio driver to configure
the encoding settings and send data over the radio.

Transmitted messages can be received by building and flashing the accompanying
LoRa receive sample on another board.
```

As soon as you add (I hope you will!) the readme for the receive sample, they should both reference one another using `` :zephyr:code-sample:`...` ``

**Review by kartben — CHANGES_REQUESTED** [review]
thank you very much @Spice-Weasel! Please see a few comments from my side, but should be good to go after that.

File: samples/drivers/lora/send/README.rst
Comment (Spice-Weasel) [reply samples/drivers/lora/send/README.rst]: This bit has been done as it CI without the space previously.

Comment (kartben) [reply samples/drivers/lora/send/README.rst]: oh yes, sorry about that, I had a pending review from earlier!

File: samples/drivers/lora/send/README.rst
Comment (Spice-Weasel) [reply samples/drivers/lora/send/README.rst:5]: Apologies, I don't understand how naming the sample with a hyphen makes it easier to find the receive sample. Could you please clarify?

File: samples/drivers/lora/send/README.rst
Comment (Spice-Weasel) [reply samples/drivers/lora/send/README.rst:14]: I'll order another board tomorrow so I should be able to do it this weekend to add the receive README.

File: samples/drivers/lora/send/README.rst
Comment (kartben) [reply samples/drivers/lora/send/README.rst:5]: > Apologies, I don't understand how naming the sample with a hyphen makes it easier to find the receive sample. Could you please clarify?

my point is that soon enough there will be a proper documentation for "LoRa receive" sample as well, right? And since it will need a unique name, I'm guessing we'll likely go for "lora-receive", so we might as well already make sure the send sample as a similar naming. Makes sense?

Comment (Spice-Weasel) [reply samples/drivers/lora/send/README.rst:5]: That's fine - I just didn't understand whether the inclusion of white-space in the name was a problem. I will make the change.

Comment (kartben) [reply samples/drivers/lora/send/README.rst:5]: Oh now I see where the confusion comes from: I suggested two different things: making the "id" unique and hyphenated, *and* changing the name so that "LoRa" is capitalized :)

**Review by kartben — APPROVED** [review]
Thank you!

In case you were wondering what's the use for the "relevant API" thingy, see how the sample is now showing here:  https://builds.zephyrproject.io/zephyr/pr/64263/docs/connectivity/lora_lorawan/index.html#id1

**Comment by Spice-Weasel** [comment]
Cool, thanks for explaining!

**Review by mniestroj — APPROVED** [review]

**Review by JordanYates — APPROVED** [review]

**Pull request merged**
